src/Hl7.Fhir.Specification.Tests/Source/ResourceResolverTests.cs

```diff
@@ -89,48 +89,50 @@ public void RetrieveWebArtifact()
             Assert.AreEqual("http://test.fhir.org/r3/StructureDefinition/Observation", artifact.GetOrigin());
         }
 
-        private class TestFhirClient : LegacyFhirClient
+        private class TestFhirClient : FhirClient
         {
             private int _status = 0;
 
             public int Status
             {
                 get { return _status; }
-                private set { _status = value; }
+                set { _status = value; }
             }
 
-            public TestFhirClient(Uri endpoint) : base(endpoint) { Status = 1; }
-
-            protected override void BeforeRequest(HttpWebRequest rawRequest, byte[] body)
-            {
-                Status = 2;
-                base.BeforeRequest(rawRequest, body);
-            }
-
-            protected override void AfterResponse(HttpWebResponse webResponse, byte[] body)
-            {
-                Status = 3;
-                base.AfterResponse(webResponse, body);
-            }
+            public TestFhirClient(string endpoint, FhirClientSettings settings = null, HttpMessageHandler handler = null) : base(endpoint, settings, handler) { Status = 1; }
         }
 
         [TestMethod, TestCategory("IntegrationTest")]
         public void RetrieveWebArtifactCustomFhirClient()
         {
-            TestFhirClient client = null;
+            using (var handler = new HttpClientEventHandler())
+            {
```

Comment:
Nested using can be without opening brace `{`. Like:
```
using (var handler = new HttpClientEventHandler())
using (var client = new TestFhirClient("http://test.fhir.org", handler: handler))
{
// etc.
}
```
or in C# 8 you can do this:
```
using var handler = new HttpClientEventHandler();
using var client = new TestFhirClient("http://test.fhir.org", handler: handler);
// etc.
```